### DISCUSSION TITLE
I'm having trouble changing My Filters

### DISCUSSION THREAD

**Discussion by McBobb** [dhowe/AdNauseam#1786] [Q&A]
I'm having a fair bit of trouble with editing my filters, both using the colour graphical interface you get when you click the **µ** button, and in the "My Filters" list directly.

First of all, I don't understand what the two columns in the graphical colour interface represent. What does clicking the green, grey and red buttons in each do?

Secondly, my main issue is confusion about why I cannot change or delete an entry in either the graphical interface, or My List once's it's been turned grey/set to **noop**.

I've tried blocking a specific ad server, specifically, akamaiedge.net, but the graphical interface button is grey, and cannot delete it from the My Filter page, or change the suffix from **noop** to **block**. Am I doing something incorrectly?

I would appreciate any help anyone might offer concerning this.

Thanks in advance.

**Comment by mneunomne** [comment]
Hello @McBobb,

Regarding the graphical interface for filtering in the Menu, this is called Dynamic Filtering, and we follow the uBlock UI in this regard. You can check more about what does these colours and columns mean [here in the uBlock wiki](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-quick-guide). 

As it is well explained in this wiki, the rules created in this menu will then appear in the "My Rules" tab, **but then I realised we do have a bug on "My rules" editor apparently, where one cannot edit or delete created filters properly**.  https://github.com/dhowe/AdNauseam/issues/1789

Luckily we are preparing a next release very soon to fix this issue as well, so keep an eye out for it!

Thanks for the feedback :)

**Reply by McBobb** [reply]
Still waiting for a new release that will fix this issue.

Is there any way to blank all the manual rules I've created for site? That might solve my problem for the time being.

Honestly, I wish there was a way to instantly undo a block I just made using the uBlock-style interface. That would be incredibly helpful. When I'm trying to click-to-block 3rd party overlays on videos, I often end up blocking the video player itself instead. Then I can't figure out any way to undo that.

**Reply by mneunomne** [reply]
Hey @McBobb , we are working on this next release still, hopefully it will be available very soon. Since we had so much merging to make with uBlock we ran into quite a few issues. But gladly the filters thing is solved at this point.

If you cannot wait for the next release to delete the rules, I think the way to undo them at the moment would be to backup your settings AND ads (if you don't want to loose your collected ads), edit your json file manually, then import it back again.

### step-by-step

- Click on `Backup from file` in the settings page. 
![Screen Shot 2021-04-16 at 23 23 12](https://user-images.githubusercontent.com/4967860/115085459-dd867d00-9f0a-11eb-804e-16184df1fee1.png)

- Select `Settings and ads` option when exporting (so you dont loose your collected ads).
![Screen Shot 2021-04-16 at 23 20 32](https://user-images.githubusercontent.com/4967860/115085331-a6b06700-9f0a-11eb-8686-b4c614199c24.png)

- open your json on the text editor and look for the parameter `dynamicFilteringString` and delete the rules that you don't want anymore and save it. 
![Screen Shot 2021-04-16 at 23 24 54](https://user-images.githubusercontent.com/4967860/115085517-f858f180-9f0a-11eb-83d6-c3a097ee5241.png)

- Click on restore from file, and select your modified json file. and voilá! your rules should be properly deleted :). 
![Screen Shot 2021-04-16 at 23 25 34](https://user-images.githubusercontent.com/4967860/115085676-335b2500-9f0b-11eb-8695-002ed1b9dd2e.png)

This is a veyr hacky solution, but if you cannot wait for the next release at the moment, this is a workaround for this unfortunate bug. 

Again thanks for your feedback,
mneunomne
